static/app/components/copyToClipboardButton.tsx

```diff
@@ -0,0 +1,69 @@
+import {ComponentProps, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button, ButtonProps} from 'sentry/components/button';
+import Clipboard from 'sentry/components/clipboard';
+import {IconCopy} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
+
+type Props = {
+  text: string;
+  iconSize?: ComponentProps<typeof IconCopy>['size'];
+} & Overwrite<
+  ButtonProps,
+  Partial<
+    Pick<ButtonProps, 'aria-label'> & {onCopy: undefined | ((copiedText: string) => void)}
```

Comment:
Should onCopy be up in Props?

Reply:
it's overriding something on the core button :(

Typescript was giving me this but I couldn't track down where the `ClipboardEventHandler` comes from.
```
Type '((value: string) => void) | undefined' is not assignable to type '(((copiedText: string) => void) & ClipboardEventHandler<any>) | undefined'.
  Type '(value: string) => void' is not assignable to type '((copiedText: string) => void) & ClipboardEventHandler<any>'.
    Type '(value: string) => void' is not assignable to type 'ClipboardEventHandler<any>'.
      Types of parameters 'value' and 'event' are incompatible.
        Type 'ClipboardEvent<any>' is not assignable to type 'string'.ts(2322)
copyToClipboardButton.tsx(13, 3): The expected type comes from property 'onCopy' which is declared here on type 'IntrinsicAttributes & { text: string; iconSize?: "xs" | "sm" | "md" | "lg" | "xl" | "xxl" | undefined; onCopy?: ((copiedText: string) => void) | undefined; } & Pick<...> & Partial<...> & { ...; } & { ...; }'
(property) onCopy?: (((copiedText: string) => void) & ClipboardEventHandler<any>) | undefined